### PR TITLE
enh(log) improve session management

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

**Fixes** MON-12197

Avoid deadlocks by improving session management in code

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

See how to test section in [jira ticket](https://centreon.atlassian.net/jira/software/c/projects/MON/boards/113?modal=detail&selectedIssue=MON-12197&sprint=654)

### CODE REVIEW

File: www/class/centreonSession.class.php [hunk 1]
```diff
@@ -75,24 +78,25 @@ public static function restart()
...
     {
         session_start();
         $_SESSION[$key] = $value;
         session_write_close();
     }
 
-    public function s_unset()
-    {
-        session_unset();
-    }
```
Comment (AssanDiomande) [inline www/class/centreonSession.class.php:88]: to be discussed

Comment (kduret) [reply www/class/centreonSession.class.php:88]: can be removed

Comment (AssanDiomande) [reply www/class/centreonSession.class.php:88]: done

File: www/class/centreonSession.class.php [hunk 2]
```diff
@@ -113,13 +107,13 @@ public static function checkSession($sessionId, CentreonDB $db)
...
+            return false;
         }
-        $prepare = $db->prepare('SELECT COUNT(*) AS total FROM session WHERE `session_id` = :session_id');
+        $prepare = $db->prepare('SELECT `session_id` FROM session WHERE `session_id` = :session_id');
         $prepare->bindValue(':session_id', $sessionId, \PDO::PARAM_STR);
         $prepare->execute();
-        $total = (int) $prepare->fetch(\PDO::FETCH_ASSOC)['total'];
-        return ($total > 0) ? 1 : 0;
+        $session = (int) $prepare->fetch(\PDO::FETCH_ASSOC)['session_id'];
+        return !empty($session) ? true : false;
```
Comment (kduret) [inline www/class/centreonSession.class.php:116]:
```suggestion
        return $prepare->fetch(\PDO::FETCH_ASSOC) !== false;
```

**Review by kduret — DISMISSED** [review]

File: www/class/centreonSession.class.php [hunk 3]
```diff
@@ -113,13 +107,12 @@ public static function checkSession($sessionId, CentreonDB $db)
         self::deleteExpiredSession($db);
 
         if (empty($sessionId)) {
-            return 0;
+            return false;
```
Comment (a-launois) [inline www/class/centreonSession.class.php:110]: docblock still says method should return an integer

**Review by a-launois — CHANGES_REQUESTED** [review]

**Review by a-launois — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**